shared-ui/src/main/java/org/hzontal/shared_ui/bottomsheet/BottomSheetUtils.kt

```diff
@@ -440,44 +447,88 @@ BottomSheetUtils {
+                .cancellable(true).screenTag("ConfirmSheet")
+        customSheetFragment.holder(
+            GenericSheetHolder(),
+            object : CustomBottomSheetFragment.Binder<GenericSheetHolder> {
+                override fun onBind(holder: GenericSheetHolder) {
+                    with(holder) {
+                        title.text = titleText
+                        description.text = descriptionText
+                        actionButtonLabel?.let {
+                            actionButton.text = it
+                        }
+                        cancelButtonLabel?.let {
+                            cancelButton.text = it
+                        }
 
-                    actionButton.setOnClickListener {
-                        consumer.accept(isConfirmed = true)
-                        customSheetFragment.dismiss()
-                    }
+                        actionButton.setOnClickListener {
+                            consumer.accept(isConfirmed = true)
+                            customSheetFragment.dismiss()
+                        }
+
+                        cancelButton.setOnClickListener {
+                            consumer.accept(isConfirmed = false)
+                            customSheetFragment.dismiss()
+                        }
+
+                        actionButton.visibility =
+                            if (actionButtonLabel.isNullOrEmpty()) View.GONE else View.VISIBLE
 
-                    cancelButton.setOnClickListener {
-                        consumer.accept(isConfirmed = false)
-                        customSheetFragment.dismiss()
                     }
+                }
+            })
+
+        customSheetFragment.transparentBackground()
+        customSheetFragment.launch()
+    }
+
+
+    fun showProgressImportSheet(
+        fragmentManager: FragmentManager,
+        titleText: String?,
+        totalProgress: Int,
+        progressNumberText: String,
+        progressStatus: MutableLiveData<Int>,
+        cancelText: String,
+        lifecycleOwner: LifecycleOwner
 
-                    actionButton.visibility =
-                        if (actionButtonLabel.isNullOrEmpty()) View.GONE else View.VISIBLE
+    ) {
+        val customSheetFragment =
+            CustomBottomSheetFragment.with(fragmentManager).page(R.layout.layout_progess_sheet)
+                .cancellable(true).statusBarColor(R.color.space_cadet)
+        customSheetFragment.holder(
+            DownloadStatustHolder(),
+            object : CustomBottomSheetFragment.Binder<DownloadStatustHolder> {
+                @RequiresApi(Build.VERSION_CODES.N)
+                override fun onBind(holder: DownloadStatustHolder) {
+                    with(holder) {
+                        progressStatus.observe(lifecycleOwner) { status ->
+                            subtitle.text = "$status/$totalProgress $progressNumberText"
```

Comment:
I think creating a model that includes all the needed field would be much more efficient `  totalProgress: Int,
        progressNumberText: String,
        progressStatus: MutableLiveData<Int>` and `statusPercent`

Reply:
I thought about adding a model but then I realised that only the progressNumberText changes  so why making a mutable livedata model with only one attribute that needs to be observed